frontend/src/components/LaunchSetting/LaunchSetting.tsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react'
+import { IService } from 'remote.it'
+import { InlineSetting } from '../InlineSetting'
+import { useApplication } from '../../services/applications'
+import { newConnection, setConnection } from '../../helpers/connectionHelper'
+
+const SSH_TYPE = 28
+
+export const LaunchSetting: React.FC<{ service: IService; connection?: IConnection }> = ({ service, connection }) => {
+  const app = useApplication(service.typeID)
+  const tokens = '[host] [port] [id]' + (service.typeID === SSH_TYPE ? ' [username]' : '')
```

Comment:
Nit: This might be more readable:

```suggestion
  let tokens = '[host] [port] [id]'
  if (service.typeID === SSH_TYPE) tokens += ' [username]'
```